components/event-bus/internal/knative/subscription/controller/subscription/reconcile.go

```diff
@@ -221,13 +215,13 @@ func (r *reconciler) deleteExternalDependency(ctx context.Context, subscription
 	}
 
 	// Check if Channel has any other Subscription, if not, delete it.
-	knativeChannel, err := r.knativeLib.GetChannel(channelName, channelNamespace)
+	knativeChannel, err := r.knativeLib.GetChannel(channelName, namespace)
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	} else if err == nil && knativeChannel != nil {
```

Comment:
Is the case where the `knativeChannel == nil` handled?

Reply:
Same for channels.
If "err == nil" then the channel object exists and it is returned.
The check "knativeChannel != nil" is not necessary

Reply:
if Knative Channel is nil, we don't need to do anything. There is nothing to delete.